src/chainlit/session.py

```diff
@@ -1,29 +1,70 @@
-from typing import Dict, TypedDict, Optional, Callable, Any, Union
+from secrets import token_urlsafe
+from typing import Dict, Optional, Callable, Any, Union
 from chainlit.client.base import BaseAuthClient, BaseDBClient
 from chainlit.types import AskResponse
 
 
-class Session(TypedDict):
+class Session:
     """Internal session object."""
 
-    # The ID of the session
-    id: str
-    # Function to ask the user a question
-    ask_user: Callable[[Any, Optional[int]], Union[AskResponse, None]]
-    # Function to emit a message to the user
-    emit: Callable[[str, Any], None]
-    # User specific environment variables. Empty if no user environment variables are required.
-    user_env: Dict[str, str]
-    # Optional langchain agent
-    agent: Any
-    # Optional llama instance
-    llama_instance: Any
-    # Whether the current task should be stopped
-    should_stop: bool
-    # Optional client to authenticate users
-    auth_client: Optional[BaseAuthClient]
-    # Optional client to persist messages and files
-    db_client: Optional[BaseDBClient]
-
-
-sessions: Dict[str, Session] = {}
+    sessions_id: Dict[str, str] = {}
+    sessions: Dict[str, "Session"] = {}
+
+    def __init__(
+        self,
+        # Session key
+        key: str,
```

Comment:
`key` and `id` are very similar abstract words and they don't make the difference between the two usages explicit. It took me quite a while to understand the difference between them even though you explained me your approach a few hours ago. 😅 What would you think of `socket_id` rather than `key`?

Reply:
The idea was to provide a Session utility not linked to the concept of websocket but as it is tightly coupled and unlikely to change, let's use `socket_id` if it clarifies the code reading :)